app/code/Magento/Wishlist/Test/Unit/Controller/Index/UpdateItemOptionsTest.php

```diff
@@ -155,93 +150,75 @@ protected function tearDown(): void
-            ->expects($this->any())
+        $this->contextMock
             ->method('getObjectManager')
-            ->willReturn($this->om);
-        $this->context
-            ->expects($this->any())
+            ->willReturn($this->objectManagerMock);
+        $this->contextMock
             ->method('getRequest')
-            ->willReturn($this->request);
-        $this->context
-            ->expects($this->any())
+            ->willReturn($this->requestMock);
+        $this->contextMock
             ->method('getEventManager')
-            ->willReturn($this->eventManager);
-        $this->context
-            ->expects($this->any())
+            ->willReturn($this->eventManagerMock);
+        $this->contextMock
             ->method('getUrl')
-            ->willReturn($this->url);
-        $this->context
-            ->expects($this->any())
+            ->willReturn($this->urlMock);
+        $this->contextMock
             ->method('getActionFlag')
             ->willReturn($actionFlag);
-        $this->context
-            ->expects($this->any())
+        $this->contextMock
             ->method('getMessageManager')
-            ->willReturn($this->messageManager);
-        $this->context->expects($this->any())
+            ->willReturn($this->messageManagerMock);
+        $this->contextMock
             ->method('getResultFactory')
             ->willReturn($this->resultFactoryMock);
     }
 
     /**
      * Get controller
      *
+     * @param bool $formKeyValid
      * @return UpdateItemOptions
      */
-    protected function getController()
+    private function getController(bool $formKeyValid = true): UpdateItemOptions
     {
         $this->prepareContext();
 
         $this->formKeyValidator->expects($this->once())
             ->method('validate')
-            ->with($this->request)
-            ->willReturn(true);
+            ->with($this->requestMock)
+            ->willReturn($formKeyValid);
 
         return new UpdateItemOptions(
-            $this->context,
-            $this->customerSession,
-            $this->wishlistProvider,
-            $this->productRepository,
+            $this->contextMock,
+            $this->customerSessionMock,
+            $this->wishlistProviderMock,
+            $this->productRepositoryMock,
             $this->formKeyValidator
         );
     }
 
-    public function testExecuteWithInvalidFormKey()
+    public function testExecuteWithInvalidFormKey(): void
     {
-        $this->prepareContext();
-
-        $this->formKeyValidator->expects($this->once())
-            ->method('validate')
-            ->with($this->request)
-            ->willReturn(false);
-
         $this->resultRedirectMock->expects($this->once())
             ->method('setPath')
             ->with('*/*/')
             ->willReturnSelf();
 
-        $controller = new Remove(
-            $this->context,
-            $this->wishlistProvider,
-            $this->formKeyValidator
-        );
```

Comment:
There was a testing of an absolutely different controller, which is a mistake. That controller is already covered with such a case.